aspnetcore/blazor/forms-validation.md

```diff
@@ -296,12 +376,69 @@ public class ShipDescription
 }
 ```
 
-::: moniker-end
+### Enable the submit button based on form validation
+
+To enable and disable the submit button based on form validation:
+
+* Use the form's `EditContext` to assign the model when the component is initialized.
+* Validate the form in the context's `OnFieldChanged` callback to enable and disable the submit button.
+
+```razor
+<EditForm EditContext="@editContext">
+    <DataAnnotationsValidator />
+    <ValidationSummary />
+
+    ...
+
+    <button type="submit" disabled="@formInvalid">Submit</button>
+</EditForm>
+
+@code {
+    private Starship starship = new Starship();
+    private bool formInvalid = true;
+    private EditContext editContext;
+
+    protected override void OnInitialized()
+    {
+        editContext = new EditContext(starship);
+
+        editContext.OnFieldChanged += (sender, e) =>
```

Comment:
```C#
editContext.OnValidationStateChanged += (_, __) =>
{
  formValid = !editContext.GetValidationMessages().Any();
  StateHasChanged();
};
```

Reply:
This doesn't quite work - fields that haven't been touched yet, but are in an invalid state, don't have validation messages yet.

Reply:
Is what I have good (for now)?

Reply:
... and I understand the `_, __` discards ... just saw those recently. Should I use those now (for the first time :tada:) for `sender` and `e`?

Reply:
... and one more ❓ ... we've been using ...

```cshtml
<EditForm Model="@model" OnValidSubmit="HandleValidSubmit">
```

`OnValidSubmit` is set to `HandleValidSubmit` throughout the topic, but would you prefer to guide readers to what you just showed ...

```cshtml
<EditForm Model="@model" OnValidSubmit="OnValidSubmit">
```

Reply:
I don't have a preference on what the event handler methods are named. What you have is fine.